proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,138 @@
+<!-- {% raw %} -->
+
+# In-process only or In-process and Out-of-process support for the compiler library
+
+* Proposal: [0004](INF-0004-inproc-outofproc-compiler-library-support.md)
```

Comment:
This proposal should start off numbered NNNN.

Reply:
(suggest leaving any renaming of this until _just before_ this PR is completed so that it doesn't confuse the diffs)

Reply:
Updated the name in the document contents.  Will update the file name when the review is close to being complete to avoid diff issues.